torchrl/objectives/returns/functional.py

```diff
@@ -195,9 +196,31 @@ def _custom_conv1d(tensor, filter):
         -1,
     )
 
-    # shape = val.shape
-    filter = filter.squeeze(-1).unsqueeze(0).unsqueeze(0)  # 1 x 1 x T
-    out = torch.conv1d(val_pad, filter)
+    if filter.ndimension() > 2:
+        # filter will have shape batch_dims x timesteps x timesteps x 1
+        # reshape to batch_dims x timesteps x 1 x 1 x timesteps ready for convolving
+        filter = filter.reshape(*filter.shape[:-2], 1, 1, filter.shape[-2])
+
+        # add dimension to val-pad as we are going to vectorise over the first
+        # axis (batch) since we have different filter per batch
+        val_pad = val_pad.unsqueeze(1)
```

Comment:
If I remove this I get `RuntimeError: wrong output shape` from [here](https://github.com/facebookresearch/rl/blob/da3e63d522c73ed7226912a18a3c5c1abeadd450/torchrl/objectives/returns/functional.py#L225-L226) 😕 

I will leave it for now as it may in any case become redundant when the other logic changes.

Reply:
This is no longer relevant with switch to `einsum`